src/message.c

```diff
@@ -1407,6 +1410,17 @@ void qd_message_send(qd_message_t *in_msg,
 
     if (msg->sent_depth < QD_DEPTH_MESSAGE_ANNOTATIONS) {
 
+        if (msg->content->aborted) {
+            // Message is aborted before any part of it has been sent.
+            // Declare the message to be sent,
+            msg->send_complete = true;
+            // the link has an outgoing deliver. abort it.
+            pn_delivery_abort(pn_link_current(pnl));
+
+            // TODO: Dispose of message buffers that may have accumulated
```

Comment:
Can we remove this TODO since the buffers are eventually freed when the message is freed?